File: tests/dart/control_flow_advanced/count_down.dart
// vybe-test: dart/control_flow_advanced/count_down
// origin: languages/dart/tests/dart/test_control_flow_advanced.rs

final StringBuffer __vybeOut = StringBuffer();

void __p(Object? o) {
  __vybeOut.writeln(o);
}

void __check(String want) {
  var got = __vybeOut.toString();
  // `writeln` on the final print contributes a trailing newline that the
  // expected line vector never carried.
  if (got.endsWith('\n')) {
    got = got.substring(0, got.length - 1);
  }
  if (got != want) {
    print('FAIL: want [$want] got [$got]');
    throw Exception('assertion failed');
  }
}

void __vybeMain() {
  var result = <int>[];
  for (var i = 5; i >= 1; i--) { result.add(i); }
  __p(result.first);
}

void main() {
  __vybeMain();
  __check('5');
}

File: tests/fortran/full_programs/count_down.f90
! vybe-test: fortran/full_programs/count_down
! origin: languages/fortran/tests/fortran/test_full_programs.rs
program t
integer :: vybe_check_i = 0
integer :: vybe_check_w(5) = [ 5, 4, 3, 2, 1 ]
integer :: i
i = 5
do while (i > 0)
vybe_check_i = vybe_check_i + 1
if (vybe_check_i > 5) then
    print *, "FAIL: more than 5 line(s)"
    stop 1
end if
if ((i) /= vybe_check_w(vybe_check_i)) then
    print *, "FAIL at ", vybe_check_i, " got [", i, "]"
    stop 1
end if
i = i - 1
end do
if (vybe_check_i /= 5) then
    print *, "FAIL: ", vybe_check_i, " line(s), wanted 5"
    stop 1
end if
end program t
